Faith_15-7_BeatThat.js

```diff
@@ -0,0 +1,220 @@
+// define global variables
+var userNameArray = [];
+var chosenNumArray = [];
+var diceArray1 = [];
+var diceArray2 = [];
+var userName;
+var diceRoll;
+var diceNum;
+var chosenNum;
+var gameMode;
+var chooseModeMsg = function () {
+  return `Please enter "lowest" or "highest" to choose your Beat That game mode.<br/>`;
+};
+var checkMode = function (input) {
+  return input == "lowest" || input == "highest";
+};
+var gameModeInstructions = function (gameMode) {
+  if (gameMode == "lowest") {
+    return `You have chosen lowest mode. Player with smallest number wins. Key in '1' to play.`;
+  }
+  if (gameMode == "highest") {
+    return `You have chosen highest mode. Player with  highest number wins. Key in '1' to play.`;
+  }
+};
+//determine largest number rolled among players by sorting array in descending order
+function largestChosenNum(diceArray, diceCt) {
+  diceArray.sort(function (a, b) {
+    return b - a;
+  });
+  var num = diceArray[0];
+  for (var i = 1; i < diceCt; i += 1) {
+    num = num * 10 + diceArray[i];
+  }
+  return num;
+}
+//determine smallest number rolled among players by sorting array in descending order
+function smallestChosenNum(diceArray, diceCt) {
+  diceArray.sort(function (a, b) {
+    return a - b;
+  });
+  var num = diceArray[0];
+  for (var i = 1; i < diceCt; i += 1) {
+    num = num * 10 + diceArray[i];
+  }
+  return num;
+}
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNum = randomInteger + 1;
+  return diceNum;
+};
+//generate n number of rolls from dice, depending how many dices you chose
+var rollDiceNum = function () {
+  for (n = 0; n < diceNum; n += 1) {
+    diceRoll = rollDice();
+    diceArray.push(diceRoll);
+  }
+};
+var resetGame = function () {
+  userNameArray = [];
+  chosenNumArray = [];
+  diceArray1 = [];
+  diceArray2 = [];
+  userName = null;
+  diceNum = null;
+  gameMode = null;
+};
+var main = function (input) {
+  //check gamemode
+  if (!gameMode) {
+    if (!input) {
+      return chooseModeMsg();
+    }
+    modeValid = checkMode(input);
+    if (!modeValid) {
+      return chooseModeMsg();
+    }
+    gameMode = input;
+    return gameModeInstructions(gameMode);
+  }
+  if (gameMode == "highest") {
+    //run game for 2 players
+    if (chosenNumArray.length >= 0 && chosenNumArray.length < 2) {
+      if (!userName) {
+        if (!input || input != 1) {
+          return `Please enter '1' to begin.`;
+        }
+        userName = input;
+        userNameArray.push(userName);
+        if (userNameArray.length == 1) {
+          return `Welcome Player ${userNameArray.length}. <br/> Enter the number of dice to roll.`;
+        } else {
+          return `Welcome Player 2. You will roll ${diceNum} dice. No input needed, click 'Submit' to continue.`;
+        }
```

Comment:
If you want to use player names to address players at lines 92 and 94, you can use a `currentPlayer` global variable to keep track of the current player and use something like `userNameArray[currentPlayer]` to get the current player's name! And this will work for multiple players as well.